### PR TITLE
STCON-67 && STCOM-150 Error Boundary

### PR DESCRIPTION
**Pull request by doytch**
STCON-67 turns off the `alert()` box when [using the `throwErrors` prop implemented here](https://github.com/folio-org/stripes-connect/pull/44). 

STCOM-150 adds [an Error Boundary at the top-level of the application.](https://reactjs.org/docs/error-boundaries.html). Error boundaries let us catch errors thrown _only_ in render() or lifecycle methods. 

React 16 unmounts the component tree when errors are thrown rather than operating in an unstable state, so error boundaries give us the opportunity to introduce a fallback UI. Contrary to what I initially thought, they are not intended to be used as a mechanism to trap errors and return to a stable state. That's why `GlobalErrorBoundary` is fairly dumb and just reports the error and doesn't do anything interesting.

### CODE REVIEW

**Comment by MikeTaylor** [comment]
To be fair, it is _in general_ not possible to recover a safe state from a problem occurring in non-pure functions. How could we write a recover function that tries to figure out what's wrong in stripes-connect's part of the Redux store? It's a beautiful dream, but I wouldn't want to be the one implementing it. (Although the simple special case of non-2xx HTTP responses is simple, of course: we just don't touch the store at all.)

File: package.json [hunk 1]
```diff
@@ -48,7 +48,7 @@
   },
```
Comment (MikeTaylor) [inline package.json]: What new stripes-connect facility are we depending on?

File: src/connectErrorEpic.js [hunk 2]
```diff
@@ -3,6 +3,9 @@ const connectErrorEpic = action$ => action$
   .map((action) => {
     const { meta, payload: e } = action;
     const op = action.type === '@@stripes-connect/FETCH_ERROR' ? 'GET' : e.type;
+
```
Comment (MikeTaylor) [inline src/connectErrorEpic.js]: Is this part of the present PR's functionality, or an unrelated change?

Comment (MikeTaylor) [reply src/connectErrorEpic.js]: Ooohhh ... maybe this is the new stripes-connect functionality that we're depending on? Great if so, but the stripes-connect `CHANGELOG.md` says nothing about what's in v3.1.2. Maybe you could fix that?

Comment (MikeTaylor) [reply src/connectErrorEpic.js]: _sigh_ save yourself some time and ignore everything I just wrote. I realised my copy of stripes-connect was outdated. I `git pull`ed and the `CHANGELOG.md` explains perfectly.

(Only one nit on that score: the entry does not reference a Jira  for `throwErrors` -- was there not one?)

**Comment by doytch** [comment]
Yeah totally, we were just originally hoping to be able to throw in render based on `failedMutations` and have error boundaries deal with the error rendering at a top level. It ain't so simple unfortunately. :)

File: src/connectErrorEpic.js (hunk 2 above)
Comment (doytch) [reply src/connectErrorEpic.js]: Ahhh, it doesn't, good nit. :) It was STCON-67

Comment (MikeTaylor) [reply src/connectErrorEpic.js]: Anyway: go ahead and merge. Thanks for handling this.

**Comment by MikeTaylor** [comment]
Jason makes an excellent point. It's good that this is merged, but we should _also_ look at isolating individual apps if possible.
